src/dstore/pmix_esh.c

```diff
@@ -742,8 +1078,9 @@ static seg_desc_t *_create_new_segment(segment_type type, char *nsname, uint32_t
             new_seg = NULL;
             PMIX_ERROR_LOG(rc);
         }
-        if (setjobuid && _is_server()){
-            if (chown(file_name, (uid_t) jobuid, (gid_t) -1) < 0){
+
+        if (_is_server() && (_setjobuid > 0)){
```

Comment:
As I stated above - we will set `_setjobuid` for clients. So you don't need to check for `_is_server()` here.